### ISSUE TITLE
Split `run-make` into two test suites: fast-path that don't need to build cargo and a slow-path that builds cargo

### ISSUE THREAD

**Issue by jieyouxu** [rust-lang/rust#134109]
Building stage 1 cargo and rustdoc takes quite a bit of time and is annoying if your run-make test doesn't even need cargo/rustdoc. Building stage 1 cargo for tests that do need cargo is actually necessary because beta cargo (the usual bootstrap cargo) might not have changes that are present in nightly cargo.

**Comment by jieyouxu** [comment]
This is likely going to want some `run_make_support` changes, such as gating the `cargo()` helpers behind a `cfg` feature that is enabled only for the slow suite.

**Comment by RalfJung** [comment]
IMO rustdoc is less of an issue than cargo since it does not require a submodule to be checked out.

**Comment by jieyouxu** [comment]
Yep. MCP is https://github.com/rust-lang/compiler-team/issues/847, the current proposal is to split `run-make` into `{run-make,run-make-cargo}`. If in the future we have further dependency / developments, we can further adjust the test suite setup as suitable.
